src/ChessBoard/MovePieces.tsx

```diff
@@ -43,22 +57,25 @@ export class MovePieces extends React.Component<Props, State> {
         return {
             x: event.clientX - left,
             y: event.clientY - top,
-        };
+        }
     }
 
     onMouseDown = (event: React.MouseEvent<HTMLCanvasElement>) => {
         const mousePositions = this.getMousePositions(event)
         const index = this.getIndexAtPosition(mousePositions.x, mousePositions.y);
         if (index !== -1) {
-            this.setState({ isDragging: true, draggingIndex: index });
+            this.isDragging = true
+            this.draggingIndex = index
+            this.initialPositions[index] = { ...this.state.positions[index] };
             window.addEventListener("pointermove", this.onMouseMove);
```

Comment:
it seems like this listeners are unnecessery or in wrong place, handlers mustn't contain any subscribtions.